packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -213,6 +213,7 @@ export default class ScriptTransformer {
             // files outside `cwd` will not be instrumented
             cwd: this._config.rootDir,
             exclude: [],
+            extension: [''],
```

Comment:
Does it have to be empty string? What empty array? Also, I wonder how this effects performance

Reply:
The empty string means “all files whose names end with the empty string”, i.e. “all files”. Looking at the [source](https://github.com/istanbuljs/test-exclude/blob/v6.0.0/index.js), there seems to be a special case for `extension: false` or `extension: []`, where the empty array would otherwise mean “no files”. It probably makes sense to write `extension: false`.

This only impacts anything if you’re using a custom transformer that supports source maps and doesn’t do its own instrumentation on a file extension not in `.js`, `.cjs`, `.mjs`, `.ts`, `.tsx`, `.jsx` with coverage enabled, i.e. the exact preconditions of the bug being fixed here. I can’t imagine someone would do that and expect Jest *not* to instrument the code. (If they do, they can simply tell Jest that their transformer does its own instrumentation and then not instrument it.)

Reply:
Yeah, this is already guarded by our own `shouldInstrument` check. We already tell istanbul to ignore its own `exclude` defaults, we should do the same for the `extension` whitelist

Reply:
If people get new files they don't want, they should use `coveragePathIgnorePatterns` on them - IMO this is somewhat of a bug fix